browser/brave_ads/device_id/device_id_impl_mac.cc

```diff
@@ -44,28 +44,44 @@ namespace {
 
 constexpr char kRootDirectory[] = "/";
 
+class MountedVolumes {
+ public:
+  MountedVolumes() { count_ = getmntinfo_r_np(&mounted_volumes_, 0); }
+  MountedVolumes(const MountedVolumes&) = delete;
+  MountedVolumes& operator=(const MountedVolumes&) = delete;
+  ~MountedVolumes() = default;
+
+  int count() const { return count_; }
+
+  base::span<struct statfs> as_span() {
+    // SAFETY: This is the nature of this API, where `count_` represents the
+    // number of elements for the `mounted_volumes_` pointer.
+    return UNSAFE_BUFFERS(base::span<struct statfs>(mounted_volumes_, count_));
+  }
+
+ private:
+  int count_ = 0;
+  struct statfs* mounted_volumes_ = nullptr;
+};
+
 // Return the BSD name (e.g. '/dev/disk1') of the root directory by enumerating
 // through the mounted volumes. Returns an empty string if an error occurred.
 std::string FindBSDNameOfSystemDisk() {
   base::ScopedBlockingCall scoped_blocking_call(FROM_HERE,
                                                 base::BlockingType::MAY_BLOCK);
 
-  struct statfs* mounted_volumes;
-  const int count = getmntinfo_r_np(&mounted_volumes, 0);
-  if (count == 0) {
+  MountedVolumes mounted_volumes;
+  if (mounted_volumes.coutn() == 0) {
```

Comment:
`count`

Reply:
yeah, sorry, I hadn't built it on mac yet, because I'm on linux.

Reply:
A sanity test? I'm not familiar with this term.